src/models/Review.js

```diff
@@ -32,4 +32,47 @@ const ReviewSchema = new mongoose.Schema(
   { timestamps: true }
 );
 
+// Create an index on 'album' and 'user' to ensure unique reviews per album-user combination
+ReviewSchema.index({ album: 1, user: 1 }, { unique: true });
+
+// Static method to calculate the average rating and number of reviews for a specific album
+ReviewSchema.statics.calculateAverageRating = async function (albumId) {
+  const result = await this.aggregate([
+    { $match: { album: albumId } }, // Match reviews with the specified albumId
+    {
+      $group: {
+        _id: null,
+        averageRating: { $avg: '$rating' },
+        numOfReviews: { $sum: 1 },
+      },
+    },
+  ]);
+
+  try {
+    await this.model('Album').findOneAndUpdate(
+      { _id: albumId }, // Find the album by its unique ObjectId.
+      {
+        // Update the 'averageRating' field of the album document with the rounded
+        // value from the aggregation result. If 'averageRating' is undefined, it defaults to 0.
+        averageRating: Math.ceil(result[0]?.averageRating || 0),
```

Comment:
`Match.ceil` will always round up. So if we have an average rating of 4.1, this will be shown as 5 stars which doesn't seem very accurate. We likely instead want [Math.round](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/round) which will round to nearest integer in either direction:
```suggestion
        averageRating: Math.round(result[0]?.averageRating || 0),
```